### PR TITLE
Add versioning spec

### PR DESCRIPTION
**Pull request by angelazhangmsft**
- Outlines CsWinRT release process and alignment with .NET release cadence
- Sets guidelines for incrementing AssemblyVersion/AssemblyFileVersion for runtime and projection assemblies
- List developer scenarios affected by version changes

### CODE REVIEW

File: docs/versioning.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+| | AssemblyFileVersion Format | 
+|-|-|
+| **WinRT.Runtime.dll** | *\<major>.\<minor>.\<patch>.\<build>* |
+| **Microsoft.Windows.SDK.NET.dll** |  *10.0.<windows_build>.<package_update>* |
+
+`AssemblyFileVersion` will be incremented with each release, regardless of whether there are API changes. This means if `AssemblyVersion` does not increment, we will still increment `AssemblyFileVersion`. The guidelines for incrementing `AssemblyFileVersion` are as follows.
+
+1. For *WinRT.Runtime.dll*:
+    * Increment *\<minor>* or *\<major>*: if there is an increment to `AssemblyVersion`. This update would be in correspondence with the update to `AssemblyVersion`.
```
Comment (Scottj1s) [inline docs/versioning.md:50]: might be clearer to state that major and minor are updated as needed (semver-ish API changes), and that AssemblyVersion is derived from them.

File: docs/versioning.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    * Increment *\<patch>* and/or *\<build>*: if there is no increment to `AssemblyVersion`.
```
Comment (Scottj1s) [inline docs/versioning.md:51]: we have yet to set patch to nonzero (but could, to be semver compliant)
build is always automatically incremented

File: docs/versioning.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+2. For *Microsoft.Windows.SDK.NET.dll*:
+    * Increment *\<package_update>*.
```
Comment (Scottj1s) [inline docs/versioning.md:53]: always automatically update package_update (unique per build)

File: docs/versioning.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+## Future Version Numbers
+
+As of .NET5 RTM (5.0.100), `AssemblyVersion` and `AssemblyFileVersion` for the runtime and SDK assemblies are identical. Going forwards, `AssemblyVersion` and `AssemblyFileVersion` will be decoupled.
+
+The CsWinRT NuGet version and SDK projection package version will correspond with the `AssemblyFileVersion` of *WinRT.Runtime.dll* and *Microsoft.Windows.SDK.NET.dll*, respectively.
+
+The table below is an example of how we will increment future version numbers. The **AssemblyVersion** and **AssemblyFileVersion** column cells list the *WinRT.Runtime.dll* version followed by the *Microsoft.Windows.SDK.NET.dll* version.
```
Comment (Scottj1s) [inline docs/versioning.md:61]: could combine the last two columns for consistency, label the result "Nuget Package version", and update the legend: The version column cells list the WinRT.Runtime.dll/CsWinRT versions followed by the Microsoft.Windows.SDK.NET.dll/SDK versions.

File: docs/versioning.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+**Note**: `AssemblyVersion` for *WinRT.Runtime.dll* will never have a non-zero *\<patch>* or *\<build>*, with the exception of .NET5 RTM.
+
+### CsWinRT NuGet Prereleases
+
+For CsWinRT NuGet prereleases between any .NET servicing or feature releases, *\<patch>* will be incremented for the *WinRT.Runtime.dll* `AssemblyFileVersion` (e.g., 1.1.0 -> 1.1.1). In general, `AssemblyVersion` will not be incremented for prereleases until just prior to a feature release. The NuGet version will have the *-prerelease* suffix, for example between the 5.0.101 and 5.0.102 servicing releases, a CsWinRT prerelease NuGet version would have the format *1.1.1-prerelease.\<build>*.
+
+## Scenarios
+
+1. **[Direct Reference]**: An app developer builds on a lower SDK version than its dependency. For example, an app developer builds with .NET5 RTM (5.0.100) and references a library built with a later SDK version (e.g., .NET 5.0.200). In this scenario, .NET 5.0.200 included an `AssemblyVersion` update.
```
Comment (Scottj1s) [inline docs/versioning.md:83]: super helpful to see the specific error messages

**Comment by Scottj1s** [comment]
I ❤ this documentation! very helpful

File: docs/versioning.md (hunk 1 above)
Comment (angelazhangmsft) [reply docs/versioning.md]: Revised `AssemblyFileVersion` rules to make it more clear

File: docs/versioning.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+As of .NET5 RTM (5.0.100), `AssemblyVersion` and `AssemblyFileVersion` for the runtime and SDK assemblies are identical. Going forwards, `AssemblyVersion` and `AssemblyFileVersion` will be decoupled.
+
+The CsWinRT NuGet version and SDK projection package version will correspond with the `AssemblyFileVersion` of *WinRT.Runtime.dll* and *Microsoft.Windows.SDK.NET.dll*, respectively.
+
+The table below is an example of how we will increment future version numbers. 
+- The **AssemblyVersion** and **AssemblyFileVersion** column cells list the *WinRT.Runtime.dll* version followed by the *Microsoft.Windows.SDK.NET.dll* version. 
+- The **NuGet package version** column cells list the CsWinRT NuGet package version followed by the SDK projection package version.
+
+| Release | AssemblyVersion | AssemblyFileVersion | NuGet package version | 
+|-|-|-|-|-|
```
Comment (Scottj1s) [inline docs/versioning.md:66]: table isn't rendering correctly anymore - probably this:
|-|-|-|-|-|

**Review by Scottj1s — APPROVED** [review]

**Comment by manodasanW** [comment]
Throughout the doc, anywhere we just say SDK, I think we should change it to Windows SDK or .NET SDK as appropriate to avoid any confusions as to what we are referring to.

File: docs/versioning.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
+# Assembly Versioning
+
+This document lists our guidelines for incrementing the assembly versions for *WinRT.Runtime.dll* and *Microsoft.Windows.SDK.NET.dll*, which are the runtime and SDK projection assemblies generated by C#/WinRT. If we increment [AssemblyVersion](https://docs.microsoft.com/dotnet/standard/assembly/versioning) for either of these assemblies, application developers may have to upgrade their SDK to the newest version and re-compile their applications. This may become painful for developers as we make frequent updates to the .NET 5.0 SDK through servicing and feature releases, so the recommendation from the .NET team is to avoid making changes to `AssemblyVersion` when possible aside from major releases.
```
Comment (manodasanW) [inline docs/versioning.md:3]: This document lists the guidelines we will be following for ...?  Do you think that might be better?

**Review by manodasanW — APPROVED** [review]

**Pull request merged**